src/linker/Linker.Steps/SubStepsDispatcher.cs

```diff
@@ -14,7 +14,7 @@ namespace Mono.Linker.Steps
 	// consist of multiple steps. It simplifies their implementation as well as the
 	// way how to hook them into the pipeline of existing steps.
 	//
-	public abstract class SubStepsDispatcher : IStep
+	public abstract class SubStepsDispatcher : IPerAssemblyStep
```

Comment:
Ouch, that might hurt that will break all Xamarin steps. I think this should be done as the new type to allow the unforced transition.

Reply:
Made it a new type instead.